sql/stats/join.go

```diff
@@ -103,22 +102,27 @@ func joinAlignedStats(left, right []sql.HistogramBucket, cmp func(sql.Row, sql.R
 		// mcvs counted in isolation
 		// todo: should we assume non-match MCVs in smaller set
 		// contribute MCV count * average frequency from the larger?
-		var mcvMatch int
-		for i, key1 := range l.Mcvs() {
-			for j, key2 := range r.Mcvs() {
-				v, err := cmp(key1, key2)
-				if err != nil {
-					return nil, err
-				}
-				if v == 0 {
-					rows += l.McvCounts()[i] * r.McvCounts()[j]
-					lRows -= float64(l.McvCounts()[i])
-					rRows -= float64(r.McvCounts()[j])
-					lDistinct--
-					rDistinct--
-					mcvMatch++
-					break
-				}
+		var mcvMatch float64
```

Comment:
Why is this a float instead of an int64? We cast it back to an int64 before using it anyway. Is this just a convoluted way of capping it at 2^(52+1) + 1 (the smallest int that can't be written as a float64)?

Reply:
good catch, just an oversight on my part. I wrote another version of this that used floats for all of the arithmetic, but that version was slower.